### DISCUSSION TITLE
Add Programmer Pickit 4

### DISCUSSION THREAD

**Discussion by GauchoMatrero** [Sloeber/arduino-eclipse-plugin#1632] [Show and tell]
**Actually this is not a bug, but a wish.**

I'm using Sloeber 4.4.1 and works fine for me, but I would like to use my Pickit4 programmer with Sloeber.

Unlike AVR Dragon that has an uncompatible USB driver, Pickit 4 works fine not only with Atmel Studio, now Microchip Studio, where it can be used for debugging.

Avrdude supports it, also Averdudess, so I think it can be  added in new Sloeber revisions. 

It's a nice feature for me using a number of attinys, and atmegas, not in the standard Arduino board.

Regards
Carlos

**Comment by jantje** [comment]
The programmers are defined in programmers.txt
If you know the commands you should be able to find out how to create/modify a programmers.txt file 
Arduino documented that here https://arduino.github.io/arduino-cli/0.32/platform-specification/#programmerstxt

**Comment by GauchoMatrero** [comment]
Thanks. I've already modified my system, but it's not standard.
I copied the last avrdude executable file and configuration to the
corresponding packages/avrdude.
and added the following lines:

pickit4_isp.name=Pickit 4 ISP (AVR)
pickit4_isp.communication=usb
pickit4_isp.protocol=pickit4_isp
pickit4_isp.program.tool=avrdude
pickit4_isp.program.tool.default=avrdude
pickit4_isp.program.protocol=pickit4_isp
#pickit4_isp.program.extra_params=-Pusb -c pickit4_isp
#pickit4_isp.usbpid = 0x2177, 0x2178, 0x2179;

To the ...avr\1.8.6\  programmers.txt programmers.sloeber.txt

This works fine, but only in the 1.8.6 arduino version, and If I install a
new version, this changes will be whipped out. That's why I suggested to
add this new and cheap programmer/debugger.
Of course this is simpler to use than an external tool un microchip studio
7 external tools.

BTW once had a AVR Dragon, and you can use it only on AVR environment, and
also i was so fragile it blew up beyond repair after a few months use.

Regards
Carlos

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Libre
de virus.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
